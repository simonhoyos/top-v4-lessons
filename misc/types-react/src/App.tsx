import React from 'react';
import logo from './logo.svg';
import './App.css';

interface Props {
  name: string
}

const Greeting: React.FC<Props> = ({ name }) => {
  return (
    <h1>Hola {name}</h1>
  );
}

// interface Props {}
// interface State {}

// class Greeting extends React.Component<Props, State> {

// }

function App() {
  return (
    <div className="App">
      <Greeting name="Simon" />
    </div>
  );
}

export default App;
