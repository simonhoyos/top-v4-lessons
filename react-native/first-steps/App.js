import React from 'react';
import { StyleSheet, Text, View, Image, TextInput, ScrollView, Button } from 'react-native';

export default function App() {
  const [value, setValue] = React.useState('hola');
  const [list, setList] = React.useState([]);

  function handleChange(e) {
    const { value } = e.target;
    setValue(value);
  }

  function handlePress() {
    const newList = [...list, value];
    console.log(newList);
    setList(newList);
    console.log(list);
  }

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Title</Text>
      <Text style={styles.text}>Open up App.js to start working on your app!</Text>
      <Text style={styles.text}>Hola Mundo</Text>
      <ScrollView style={{ height: 250, backgroundColor: '#fff' }}>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquid ut, vero laborum tempore doloremque voluptatibus cum nisi maxime neque quibusdam eligendi nostrum quod laboriosam vel commodi ad hic, aliquam exercitationem!</Text>
      </ScrollView>
      {/* <Image source="https://reactnative.dev/docs/assets/p_cat2.png" /> */}
      <Image source="https://reactnative.dev/docs/assets/p_cat2.png" style={{width: 200, height: 200}}/>
      <TextInput style={styles.input} value={value} onChange={handleChange} />
      <Button onPress={handlePress} title="Submit"/>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'purple',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: '#fff',
  },
  title: {
    color: '#fff',
    fontSize: 32,
    fontWeight: 'bold'
  },
  input: {
    backgroundColor: '#fff'
  }
});
