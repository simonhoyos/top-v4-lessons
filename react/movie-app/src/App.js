import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Link,
  Redirect
} from 'react-router-dom';
import Popular from './pages/Popular';
import Upcoming from './pages/Upcoming';
import TopRated from './pages/TopRated';
import NowPlaying from './pages/NowPlaying';
import { MoviesProvider } from './state/MoviesContext';
import './App.css';

function App() {
  return (
    <MoviesProvider>
      <Router>
        <nav>
          <Link to="/popular">Popular</Link>
          <Link to="/upcoming">Upcoming</Link>
          <Link to="/top-rated">Top Rated</Link>
          <Link to="/now-playing">Now Playing</Link>
        </nav>
        <div className="App">
          <Switch>
            <Route exact path="/popular" component={Popular} />
            <Route exact path="/upcoming" component={Upcoming} />
            <Route exact path="/top-rated" component={TopRated} />
            <Route exact path="/now-playing" component={NowPlaying} />
            <Redirect path="*" to="/popular" />
          </Switch>
        </div>
      </Router>
    </MoviesProvider>
  );
}

export default App;
