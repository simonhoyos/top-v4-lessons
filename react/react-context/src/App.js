import React from 'react';
import logo from './logo.svg';
import './App.css';
import TasksContext from './state/TasksContext';

// lifting state
// prop drilling

// Render Props
function FatherComponent({ render }) {
  const [list, setList] = React.useState(['hola mundo', 'hola mir']);
  return render({ list });
}

// class TasksList extends React.Component {
//   static contextType = TasksContext;
//   render() {
//     return (
      // <div className="tasks">
      //   {this.context.tasks.map(task => <p>{task}</p>)}
      // </div>
//     );
//   }
// }

function TasksList() {
  // const { tasks } = React.useContext(TasksContext);

  // return (
    // <div className="tasks">
    //   {tasks && tasks.length > 0 && tasks.map(task => <p>{task}</p>)}
    // </div>
  // );

  return (
    <TasksContext.Consumer>
      {({ tasks }) => (
        <div className="tasks">
          {tasks && tasks.length > 0 && tasks.map(task => <p>{task}</p>)}
        </div>
      )}
    </TasksContext.Consumer>
  );
}

function Button() {
  const { addTask } = React.useContext(TasksContext);

  return <button onClick={addTask}>Agregar tarea</button>;
}

function App() {
  const [tasks, setTasks] = React.useState([]);

  function addTask() {
    setTasks([ ...tasks, 'nueva tarea' ]);
  }

  return (
    <TasksContext.Provider value={{ tasks: ['otro contexto'] }}>
      <TasksList />
      <TasksContext.Provider value={{ tasks, addTask }}>
        <div className="App">
          <TasksList />
          <Button />
          <FatherComponent render={
            ({ list }) => {
              return list.map(el => <p>{el}</p>);
            }
          } />
        </div>
      </TasksContext.Provider>
    </TasksContext.Provider>
  );
}

export default App;
