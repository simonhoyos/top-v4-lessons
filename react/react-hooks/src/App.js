import React from 'react';
import axios from 'axios';
import './App.css';

// class App extends React.Component {
//   state = {
//     count: 0
//   };

//   handleClick = () => {
//     this.setState((prevState) => ({ count: prevState.count + 1 }));
//   }

//   render() {
//     const { count } = this.state;
//     return (
      // <div className="App">
      //   <button onClick={this.handleClick}>{count}</button>
      // </div>
//     );
//   }
// }

function App() {
  const [count, setCount] = React.useState(0);
  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState(null);
  const [posts, setPosts] = React.useState([]);

  React.useEffect(() => {
    setLoading(true);
    axios({
      method: 'GET',
      baseURL: 'http://jsonplaceholder.typicode.com',
      url: '/posts'
    })
      .then(({ data }) => setPosts(data))
      .catch((error) => setError(error))
      .finally(() => setLoading(false));
  }, []);

  function handleClick() {
    setCount(count + 1);
  }

  if(loading) return <p>Loading...</p>;
  if(error) return <p>Something went wrong</p>;
  return (
    <div className="App">
      <button onClick={handleClick}>{count}</button>
      {posts && posts.length && posts.map(({ id, title, body}) => (
        <div key={id} className="post">
          <h2>{title}</h2>
          <p>{body}</p>
        </div>
      ))}
    </div>
  );
}

export default App;
