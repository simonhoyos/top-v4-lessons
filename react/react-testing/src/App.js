import React from 'react';
import TasksList from './components/TasksList';
import TaskForm from './components/TaskForm';
import './App.css';

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <TasksList />
        <TaskForm />
      </div>
    );
  }
}

export default App;
