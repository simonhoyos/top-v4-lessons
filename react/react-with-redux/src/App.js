import React from 'react';
import { connect } from 'react-redux';
import { increaseCount, decreaseCount } from './reducers/countReducer';
import Posts from './components/Posts';
import './App.css';

function App({
  count,
  increaseCount,
  decreaseCount,
}) {
  return (
    <div className="App">
      <h1>{count}</h1>
      <button
        type="button"
        onClick={increaseCount}
      >
        Increase
      </button>
      <button
        type="button"
        onClick={decreaseCount}
      >
        Decrease
      </button>
      <Posts />
    </div>
  );
}

// HOC - High Order Component
// HOF - High Order Function

const mapStateToProps = ({ countReducer }) => {
  return {
    count: countReducer.count,
  }
}

const mapDispatchToProps = {
  increaseCount,
  decreaseCount,
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
