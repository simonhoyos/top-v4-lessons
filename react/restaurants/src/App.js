import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { Signup } from './pages/Signup';
import { Home } from './pages/Home';
import './App.css';

function PrivateRoute(props) {
  const token = localStorage.token;

  if(!token) return <Redirect to="/signup" />
  return (
    <Route {...props} />
  );
}

function App() {
  return (
    <div className="App">
      <Router>
        <Switch>
          <Route exact path="/admin/signup" component={Signup} />
          <PrivateRoute exact path="/" component={Home} />
          <Redirect from="*" to="/admin/signup" />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
