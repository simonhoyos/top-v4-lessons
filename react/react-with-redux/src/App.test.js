import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import App from './App';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('App' , () => {
  it('should match snapshot', () => {
    const store = mockStore({
      increaseCount: jest.fn(),
      decreaseCount: jest.fn(),
      getPosts: jest.fn(),
      countReducer: { count: 0 },
      postsReducer: { posts: [] },
    });

    const wrapper = mount(
      <Provider store={store}>
        <App store={store} />
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
