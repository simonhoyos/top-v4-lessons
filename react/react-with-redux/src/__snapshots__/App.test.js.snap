// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <App
      count={0}
      decreaseCount={[Function]}
      increaseCount={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <div
        className="App"
      >
        <h1>
          0
        </h1>
        <button
          onClick={[Function]}
          type="button"
        >
          Increase
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Decrease
        </button>
        <Connect(Posts)>
          <Posts
            getPosts={[Function]}
            posts={Array []}
          >
            <div
              className="posts"
            />
          </Posts>
        </Connect(Posts)>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;
